# # Question

Q1.Analysie Account dataframe
Q2. track down active deals






In [1]:
import numpy as np
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('float_format', '{:f}'.format)

In [2]:
#Loading Dataset
accounts=pd.read_csv('Dataset/Accounts.csv')
account_properties=pd.read_csv('Dataset/Accounts_properties.csv')
deals=pd.read_csv('Dataset/Deals_to_Properties.csv')
opportunities =pd.read_csv('Dataset/Opportunities.csv')
properties = pd.read_csv('Dataset/Properties.csv')
test_data= pd.read_csv('Dataset/Test_Data.csv')

In [3]:
test_data

id_accs
0   0012A00002AXHJUQA5
1   0012A000023XlWaQAK
2   0012A000023Y4bYQAS
3   0012A000023XlbuQAC
4   0012A000023Xlh9QAC
5   0012A000023XlhdQAC
6   0012A000023XlkNQAS
7   0012A000023XlnWQAS
8   0012A000023XlnqQAC
9   0012A000023Xp5LQAS
10  0012A000023Xm35QAC
11  0012A000023YB67QAG
12  0012A000023XmRwQAK
13  0012A000023XmmFQAS
14  0012A000023Y1rtQAC
15  0012A000023YBFRQA4
16  0012A000023Xo0OQAS
17  0012A000023XoRjQAK
18  0012A000023XwMRQA0
19  0012A000029ZzX1QAK
20  0012A000023YLFWQA4
21  0012A000023Xp4LQAS
22  0012A000023XrTNQA0
23  0012A00002JF3YmQAL
24  0012A000023YKsuQAG
25  0012A000023Y9gNQAS
26  0012A000026WwqxQAC
27  0012A000023YAupQAG
28  0012A000023YDnWQAW

# Data Analysis

In [46]:
accounts.to_csv('accounts.csv')

In [5]:
accounts.isnull().sum()

id_accs                           0
active_deals                      0
activity_count                    0
buyer_book                        0
servicing_contract                0
investor_type                    21
cmbs                              0
consultant                        0
correspondent                     0
foreign                           0
master_servicer                   0
lender_book                       0
loan_sales_book                   0
loan_servicing                    0
num_deals_as_client               0
num_deals_as_investor             0
number_of_properties              0
number_of_related_deals           0
number_of_related_properties      0
number_of_won_deals_as_client     0
dtype: int64

In [6]:
account_properties.count()

id          70350
id_deals    70350
id_accs     70350
id_props    70350
dtype: int64

In [7]:
account_properties.isnull().sum()

id          0
id_deals    0
id_accs     0
id_props    0
dtype: int64

In [8]:
deals.count()

id             25820
createdbyid    25820
id_deals       25820
id_props       25820
dtype: int64

In [9]:
deals.isnull().sum()

id             0
createdbyid    0
id_deals       0
id_props       0
dtype: int64

In [10]:
properties.count()

id_props                 18298
building_status          10081
building_tax_expenses     9555
id_deals                 18298
city                     18298
class                    11350
county                   18298
sale_status              18105
portfolio                18298
market                   18297
num_buildings            10748
num_floors               13626
num_parking_spaces        7830
occupancy_rate           14246
price_per_sq_ft          11861
price_per_unit            5677
property_type_1          17987
region__c                18293
sale_amount__c           11858
sale_date__c             12205
size_acres               13944
size_sf                  16730
size_units                8239
year_built               16440
dtype: int64

In [11]:
properties.isnull().sum()

id_props                     0
building_status           8217
building_tax_expenses     8743
id_deals                     0
city                         0
class                     6948
county                       0
sale_status                193
portfolio                    0
market                       1
num_buildings             7550
num_floors                4672
num_parking_spaces       10468
occupancy_rate            4052
price_per_sq_ft           6437
price_per_unit           12621
property_type_1            311
region__c                    5
sale_amount__c            6440
sale_date__c              6093
size_acres                4354
size_sf                   1568
size_units               10059
year_built                1858
dtype: int64

In [12]:
opportunities.count()

id_deals            26029
id_accs             26029
accounting_date     25516
best_initial_bid    25028
id_deals.1          26029
closedate           26029
date_closed         25525
deal_type           26023
deal_update_flag    26029
debt_yield          13851
fiscal              26029
fiscalquarter       26029
fiscalyear          26029
platform            26023
property_group      26027
property_type       26027
dtype: int64

In [13]:
opportunities.isnull().sum()

id_deals                0
id_accs                 0
accounting_date       513
best_initial_bid     1001
id_deals.1              0
closedate               0
date_closed           504
deal_type               6
deal_update_flag        0
debt_yield          12178
fiscal                  0
fiscalquarter           0
fiscalyear              0
platform                6
property_group          2
property_type           2
dtype: int64

In [14]:
opportunities.head()

id_deals             id_accs accounting_date  best_initial_bid  \
0  0062A00000moF1LQAU  0012A000023XvIOQA0             NaN               nan   
1  0062A00000moF9LQAU  0012A000023YFe1QAG             NaN               nan   
2  0062A00000moGodQAE  0012A000023YEDcQAO       8/12/2004               nan   
3  0062A00000moBTHQA2  0012A000023YA3nQAG             NaN               nan   
4  0062A00000mo8OjQAI  0012A000023XpRUQA0             NaN               nan   

           id_deals.1   closedate      date_closed          deal_type  \
0  0062A00000moF1LQAU   8/12/2005   8/18/2005 0:00         Fixed-rate   
1  0062A00000moF9LQAU    2/6/2012  6/27/2012 12:00         Fixed-rate   
2  0062A00000moGodQAE  11/24/2003   8/11/2004 0:00  Construction Loan   
3  0062A00000moBTHQA2   2/10/2006   4/19/2006 0:00         Fixed-rate   
4  0062A00000mo8OjQAI   7/13/2006  11/16/2006 0:00         Fixed-rate   

  deal_update_flag  debt_yield  fiscal  fiscalquarter  fiscalyear platform  \
0             True         nan  2005 3              3        2005     Debt   
1             True         nan  2012 1              1        2012     Debt   
2             True         nan  2003 4              4        2003     Debt   
3             True         nan  2006 1              1        2006     Debt   
4             True         nan  2006 3              3        2006     Debt   

  property_group property_type  
0     Healthcare    Healthcare  
1     Healthcare       Biotech  
2     Healthcare       Biotech  
3     Healthcare    Healthcare  
4     Healthcare    Healthcare

In [15]:
accounts.ftypes

id_accs                          object:dense
active_deals                      int64:dense
activity_count                    int64:dense
buyer_book                         bool:dense
servicing_contract                 bool:dense
investor_type                    object:dense
cmbs                               bool:dense
consultant                         bool:dense
correspondent                      bool:dense
foreign                            bool:dense
master_servicer                    bool:dense
lender_book                        bool:dense
loan_sales_book                    bool:dense
loan_servicing                     bool:dense
num_deals_as_client               int64:dense
num_deals_as_investor             int64:dense
number_of_properties              int64:dense
number_of_related_deals           int64:dense
number_of_related_properties      int64:dense
number_of_won_deals_as_client     int64:dense
dtype: object

In [16]:
investor_type= accounts.investor_type.unique()
investor_type_mapping = {investor_type[row]:row for row in range(len(investor_type))}

In [17]:
accounts.investor_type=accounts.investor_type.apply(lambda x :investor_type_mapping[x])

In [18]:
accounts.head(10)

id_accs  active_deals  activity_count buyer_book  \
0  0012A000023XlCCQA0             1            2367       True   
1  0012A000023XmOxQAK             9            7974       True   
2  0012A000023YLqwQAG           891           50161       True   
3  0012A000023XlNjQAK            34           14631       True   
4  0012A000023YBFRQA4           283          194195       True   
5  0012A000023XlQFQA0            39           23659       True   
6  0012A000023XlWaQAK           261          127948       True   
7  0012A000029wI3eQAE            14           23727       True   
8  0012A000023XmSVQA0            11            2405       True   
9  0012A00002HqXOHQA3           372            5954       True   

  servicing_contract  investor_type   cmbs consultant correspondent foreign  \
0              False              0  False      False         False   False   
1              False              1  False      False         False   False   
2               True              2  False      False         False   False   
3              False              3   True      False         False    True   
4              False              4  False      False         False   False   
5              False              5  False      False         False   False   
6              False              4  False      False         False   False   
7              False              0  False      False         False   False   
8              False              0  False      False         False   False   
9               True              3   True      False         False   False   

  master_servicer lender_book loan_sales_book loan_servicing  \
0           False       False           False          False   
1           False        True            True          False   
2           False        True            True          False   
3           False        True            True          False   
4           False        True            True          False   
5           False       False            True          False   
6           False        True            True          False   
7           False        True            True          False   
8           False       False           False          False   
9           False        True            True          False   

   num_deals_as_client  num_deals_as_investor  number_of_properties  \
0                    0                      1                     1   
1                    7                      2                    77   
2                   40                    851                   693   
3                    0                     34                     0   
4                   63                    220                   393   
5                   35                      4                    57   
6                   40                    221                   138   
7                    9                      5                    64   
8                   10                      1                     2   
9                    2                    370                     0   

   number_of_related_deals  number_of_related_properties  \
0                        2                             1   
1                       18                            85   
2                      996                          1167   
3                       45                            14   
4                      383                           591   
5                       64                            58   
6                      283                           276   
7                       40                            71   
8                       16                             2   
9                        0                           164   

   number_of_won_deals_as_client  
0                              0  
1                              7  
2                             37  
3                              0  
4                             46  
5                             32  
6                       

In [19]:
account_properties.ftypes

id          object:dense
id_deals    object:dense
id_accs     object:dense
id_props    object:dense
dtype: object

In [20]:
accounts_opportunities = pd.merge(accounts, opportunities, on='id_accs', how='inner')

In [21]:
accounts_opportunities.head()

id_accs  active_deals  activity_count buyer_book  \
0  0012A000023XmOxQAK             9            7974       True   
1  0012A000023XmOxQAK             9            7974       True   
2  0012A000023XmOxQAK             9            7974       True   
3  0012A000023XmOxQAK             9            7974       True   
4  0012A000023XmOxQAK             9            7974       True   

  servicing_contract  investor_type   cmbs consultant correspondent foreign  \
0              False              1  False      False         False   False   
1              False              1  False      False         False   False   
2              False              1  False      False         False   False   
3              False              1  False      False         False   False   
4              False              1  False      False         False   False   

  master_servicer lender_book loan_sales_book loan_servicing  \
0           False        True            True          False   
1           False        True            True          False   
2           False        True            True          False   
3           False        True            True          False   
4           False        True            True          False   

   num_deals_as_client  num_deals_as_investor  number_of_properties  \
0                    7                      2                    77   
1                    7                      2                    77   
2                    7                      2                    77   
3                    7                      2                    77   
4                    7                      2                    77   

   number_of_related_deals  number_of_related_properties  \
0                       18                            85   
1                       18                            85   
2                       18                            85   
3                       18                            85   
4                       18                            85   

   number_of_won_deals_as_client            id_deals accounting_date  \
0                              7  0062A00000moOejQAE       7/31/2014   
1                              7  0062A00000nBEazQAG       5/24/2017   
2                              7  0062A00000moOeeQAE       4/20/2015   
3                              7  0062A00000moPPVQA2       3/26/2015   
4                              7  0062A00000moOY4QAM        5/5/2014   

   best_initial_bid          id_deals.1   closedate     date_closed  \
0               nan  0062A00000moOejQAE    6/3/2014  7/31/2014 0:00   
1          0.000000  0062A00000nBEazQAG  12/19/2016  5/24/2017 0:00   
2          0.000000  0062A00000moOeeQAE  12/22/2014  4/17/2015 0:00   
3          0.000000  0062A00000moPPVQA2  12/15/2014  3/26/2015 0:00   
4          0.000000  0062A00000moOY4QAM   2/11/2014   5/1/2014 0:00   

       deal_type deal_update_flag  debt_yield  fiscal  fiscalquarter  \
0  Floating-Rate             True    8.700000  2014 2              2   
1  Floating-Rate             True    0.000000  2016 4              4   
2  Floating-Rate             True    9.500000  2014 4              4   
3  Floating-Rate             True    0.000000  2014 4              4   
4     Fixed-rate             True    9.250000  2014 1              1   

   fiscalyear platform property_group  property_type  
0        2014     Debt  Hotel-Lodging  Hotel/Lodging  
1        2016     Debt  Hotel-Lodging  Hotel/Lodging  
2        2014     Debt  Hotel-Lodging  Hotel/Lodging  
3        2014     Debt  Hotel-Lodging  Hotel/Lodging  
4        2014     Debt  Hotel-Lodging  Hotel/Lodging

In [22]:
accounts_opportunities.groupby('property_type')

In [23]:
accounts_opportunities_Senior = accounts_opportunities[accounts_opportunities['property_type']=='Seniors Housing']
accounts_opportunities_Day = accounts_opportunities[accounts_opportunities['property_type']=='Day Care']

accounts_opportunities_Storage = accounts_opportunities[accounts_opportunities['property_type']=='Self Storage']
accounts_opportunities_Apartments = accounts_opportunities[accounts_opportunities['property_type']=='Apartments']
accounts_opportunities_Industrial = accounts_opportunities[accounts_opportunities['property_type']=='Industrial']
accounts_opportunities_Hotel = accounts_opportunities[accounts_opportunities['property_type']=='Hotel/Lodging']
accounts_opportunities_Office = accounts_opportunities[accounts_opportunities['property_type']=='Office']
accounts_opportunities_Land = accounts_opportunities[accounts_opportunities['property_type']=='Land']
accounts_opportunities_Biotech = accounts_opportunities[accounts_opportunities['property_type']=='Biotech']
accounts_opportunities_Retail = accounts_opportunities[accounts_opportunities['property_type']=='Retail']
accounts_opportunities_Data = accounts_opportunities[accounts_opportunities['property_type']=='Data Center']
accounts_opportunities_Healthcare = accounts_opportunities[accounts_opportunities['property_type']=='Healthcare']
accounts_opportunities_MHC = accounts_opportunities[accounts_opportunities['property_type']=='Manufactured Home Communities']
accounts_opportunities_Recreational = accounts_opportunities[accounts_opportunities['property_type']=='Recreational/Speciality']
accounts_opportunities_parking = accounts_opportunities[accounts_opportunities['property_type']=='Parking']
accounts_opportunities_Mixed = accounts_opportunities[accounts_opportunities['property_type']=='Mixed Use']
accounts_opportunities_SFR = accounts_opportunities[accounts_opportunities['property_type']=='SFR - Attached']
accounts_opportunities_Promissory  = accounts_opportunities[accounts_opportunities['property_type']=='Promissory Note']

In [24]:
accounts_opportunities_Apartments.groupby('investor_type').size()


investor_type
0     2304
1      229
2      102
3        9
4       41
5      863
6       34
7       20
10     145
11      13
12      11
13       4
15      14
dtype: int64

In [25]:
investor_type_mapping_reverse = {value : key for key,value in investor_type_mapping.items()}

In [26]:

investor_type_mapping_reverse

{0: 'Private Investor/Developer',
 1: 'REIT',
 2: 'Bank / Thrift',
 3: 'Conduit/CMBS',
 4: 'Life Insurance Company',
 5: 'Advisor/Equity Fund',
 6: 'Credit/Specialty Finance Company',
 7: 'Agency',
 8: 'Property Owner',
 9: 'Account',
 10: 'Corporate Investor/User',
 11: 'Unknown',
 12: 'Pension Fund',
 13: nan,
 14: 'Sovereign Wealth Fund',
 15: 'Special Servicer',
 16: 'Endowment',
 17: 'Property Lender',
 18: 'Tenants in Common'}

In [27]:
accounts_opportunities.columns

Index(['id_accs', 'active_deals', 'activity_count', 'buyer_book',
       'servicing_contract', 'investor_type', 'cmbs', 'consultant',
       'correspondent', 'foreign', 'master_servicer', 'lender_book',
       'loan_sales_book', 'loan_servicing', 'num_deals_as_client',
       'num_deals_as_investor', 'number_of_properties',
       'number_of_related_deals', 'number_of_related_properties',
       'number_of_won_deals_as_client', 'id_deals', 'accounting_date',
       'best_initial_bid', 'id_deals.1', 'closedate', 'date_closed',
       'deal_type', 'deal_update_flag', 'debt_yield', 'fiscal',
       'fiscalquarter', 'fiscalyear', 'platform', 'property_group',
       'property_type'],
      dtype='object')

 ALL THE DATA RESIDES HERE 
 Total accounts we have = 2756
 Historical Data available for deals=  1179
 total number or properties = 18298

len(accounts_opportunities['id_accs'].unique()) #1179

In [28]:
accounts_opportunities_right = pd.merge(accounts, opportunities, on='id_accs', how='inner')

In [29]:
accounts_opportunities_propeties = pd.merge(accounts_opportunities_right, deals, on='id_deals', how='inner')

In [30]:
accounts_opportunities_properties=pd.merge(accounts_opportunities_propeties, properties, on='id_props', how='inner')

In [31]:
accounts_opportunities_properties.count()

id_accs                          9358
active_deals                     9358
activity_count                   9358
buyer_book                       9358
servicing_contract               9358
investor_type                    9358
cmbs                             9358
consultant                       9358
correspondent                    9358
foreign                          9358
master_servicer                  9358
lender_book                      9358
loan_sales_book                  9358
loan_servicing                   9358
num_deals_as_client              9358
num_deals_as_investor            9358
number_of_properties             9358
number_of_related_deals          9358
number_of_related_properties     9358
number_of_won_deals_as_client    9358
id_deals_x                       9358
accounting_date                  9188
best_initial_bid                 8902
id_deals.1                       9358
closedate                        9358
date_closed                      9251
deal_type   

In [32]:
grouped_by = accounts_opportunities_properties.groupby('id_accs').groups
large_customers=[]
small_customers=[]


for cust,count in grouped_by.items():
    if len(count) >2:
        large_customers.append(cust)
    else:
        small_customers.append(cust)
print("large",len(large_customers))
print("small",len(small_customers))


large 616
small 462


In [33]:
null=[(x,accounts_opportunities_properties[x].isnull().sum()) for x in accounts_opportunities_properties.columns if accounts_opportunities_properties[x].isnull().sum()>0]
null

[('accounting_date', 170),
 ('best_initial_bid', 456),
 ('date_closed', 107),
 ('debt_yield', 1623),
 ('building_status', 3495),
 ('building_tax_expenses', 3711),
 ('class', 2791),
 ('market', 1),
 ('num_buildings', 2687),
 ('num_floors', 1496),
 ('num_parking_spaces', 4679),
 ('occupancy_rate', 1058),
 ('price_per_sq_ft', 2082),
 ('price_per_unit', 5849),
 ('property_type_1', 51),
 ('region__c', 3),
 ('sale_amount__c', 1833),
 ('sale_date__c', 1738),
 ('size_acres', 1409),
 ('size_sf', 382),
 ('size_units', 5083),
 ('year_built', 432)]

In [34]:
properties_unique= accounts_opportunities_properties['id_props'].unique()



In [35]:
properties_intrested = properties.where(properties['id_props'].isin(list(properties_unique)))

In [36]:
properties_intrested=properties_intrested[properties_intrested['id_props'].isnull()==False]
properties_intrested.head()

id_props building_status     building_tax_expenses  \
0   a0I2A00000XQGKlUAP        Existing       2009 Tax @ $1.25/sf   
3   a0I2A00000XQGKpUAP             NaN                       NaN   
5   a0I2A00000XQGKrUAP        Existing       2016 Tax @ $5.42/sf   
6   a0I2A00000XQGKsUAP        Existing  2015 Tax @ $1158.52/Unit   
10  a0I2A00000XQGKxUAP             NaN                       NaN   

              id_deals       city    class       county sale_status  \
0   a0I2A00000XQGKlUAP  City-0488  Class B  County-0172           N   
3   a0I2A00000XQGKpUAP  City-1415  Class A  County-0452           N   
5   a0I2A00000XQGKrUAP  City-0210  Class A  County-0613           N   
6   a0I2A00000XQGKsUAP  City-0649  Class B  County-0624           N   
10  a0I2A00000XQGKxUAP  City-0283      NaN  County-0404           N   

    portfolio       market  num_buildings  num_floors  num_parking_spaces  \
0    0.000000  Market-0059       1.000000    1.000000           56.000000   
3    0.000000  Market-0165       1.000000   25.000000                 nan   
5    0.000000  Market-0032       1.000000    1.000000          793.000000   
6    0.000000  Market-0059      13.000000    2.000000          300.000000   
10   0.000000  Market-0149            nan         nan                 nan   

    occupancy_rate  price_per_sq_ft  price_per_unit property_type_1  \
0        95.000000        46.590000 18350000.000000      Industrial   
3              nan       580.830000  1220689.660000             NaN   
5       100.000000       245.400000             nan          Office   
6        95.000000        73.900000    59134.620000   Multi-Housing   
10      100.000000              nan             nan      Industrial   

    region__c   sale_amount__c sale_date__c  size_acres       size_sf  \
0   Southwest  18350000.000000    7/10/2015    3.760000 393862.000000   
3   Northeast 141600000.000000    1/21/2016    0.530000 243789.000000   
5   Northeast  40000000.000000    5/30/2012    0.460000 163000.000000   
6   Southwest  12300000.000000    2/24/2015    9.630000 166432.000000   
10  Southeast              nan          NaN         nan 513734.000000   

    size_units  year_built  
0     1.000000 1980.000000  
3   116.000000         nan  
5          nan 1937.000000  
6   208.000000 1979.000000  
10         nan 2011.000000

In [37]:
large_customers_accounts = accounts_opportunities_properties.where(accounts_opportunities_properties['id_accs'].isin(large_customers))
large_customers_accounts=large_customers_accounts[large_customers_accounts['id_accs'].isnull()==False]
prop =map(int,large_customers_accounts['investor_type'].unique())
for val in prop:
    print(val ,investor_type_mapping_reverse[val])
    

1 REIT
5 Advisor/Equity Fund
2 Bank / Thrift
0 Private Investor/Developer
4 Life Insurance Company
10 Corporate Investor/User
6 Credit/Specialty Finance Company
3 Conduit/CMBS
11 Unknown
12 Pension Fund
9 Account
13 nan
15 Special Servicer
7 Agency


In [38]:
small_customers_accounts = accounts_opportunities_properties.where(accounts_opportunities_properties['id_accs'].isin(small_customers))
small_customers_accounts=small_customers_accounts[small_customers_accounts['id_accs'].isnull()==False]
prop =map(int,small_customers_accounts['investor_type'].unique())
for val in prop:
    print(val ,investor_type_mapping_reverse[val])
    
    
#Special servicer and Account will come in large only and sovereign wealth fund in small only

0 Private Investor/Developer
3 Conduit/CMBS
5 Advisor/Equity Fund
7 Agency
2 Bank / Thrift
10 Corporate Investor/User
1 REIT
11 Unknown
6 Credit/Specialty Finance Company
4 Life Insurance Company
12 Pension Fund
14 Sovereign Wealth Fund
13 nan


In [39]:
account_riet = accounts_opportunities_properties[accounts_opportunities_properties['investor_type']==1]
account_riet.describe()


/home/akash_teehnoge/anaconda3/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


active_deals  activity_count  investor_type  num_deals_as_client  \
count   1178.000000     1178.000000    1178.000000          1178.000000   
mean      49.584890    30173.888795       1.000000            38.597623   
std       36.315782    23001.587154       0.000000            33.173513   
min        2.000000      671.000000       1.000000             1.000000   
25%       19.000000    13087.000000       1.000000            14.000000   
50%       39.000000    20762.000000       1.000000            29.000000   
75%       72.000000    45671.750000       1.000000            55.000000   
max      116.000000   109059.000000       1.000000           110.000000   

       num_deals_as_investor  number_of_properties  number_of_related_deals  \
count            1178.000000           1178.000000              1178.000000   
mean               10.987267            182.443124                82.198642   
std                17.205954            240.327104                57.885610   
min                 0.000000              1.000000                 2.000000   
25%                 3.000000             27.000000                28.000000   
50%                 5.000000             89.000000                68.000000   
75%                 9.000000            278.000000               144.500000   
max                83.000000           1124.000000               179.000000   

       number_of_related_properties  number_of_won_deals_as_client  \
count                   1178.000000                    1178.000000   
mean                     202.029711                      34.254669   
std                      249.305605                      31.637115   
min                        1.000000                       1.000000   
25%                       52.250000                      10.000000   
50%                      104.000000                      22.000000   
75%                      297.000000                      52.000000   
max                     1166.000000                     103.000000   

       best_initial_bid  debt_yield  fiscalquarter  fiscalyear  num_buildings  \
count       1128.000000  992.000000    1178.000000 1178.000000     747.000000   
mean     8633513.026596    0.526613       2.556027 2011.218166       5.305221   
std     30934415.935541    2.284553       1.123928    3.746528      10.003780   
min            0.000000    0.000000       1.000000 2002.000000       0.000000   
25%                 nan         nan       2.000000 2008.000000            nan   
50%                 nan         nan       3.000000 2012.000000            nan   
75%                 nan         nan       4.000000 2014.000000            nan   
max    335000000.000000   16.000000       4.000000 2016.000000      95.000000   

       num_floors  num_parking_spaces  occupancy_rate   price_per_sq_ft  \
count  980.000000          586.000000     1108.000000       1035.000000   
mean     4.460204          476.923208       88.903430    1117420.208773   
std      7.597473          717.840043       20.054232   34527769.703707   
min      0.000000            1.000000        0.000000          2.420000   
25%           nan                 nan             nan               nan   
50%           nan                 nan             nan               nan   
75%           nan                 nan             nan               nan   
max     57.000000        12000.000000      100.000000 1110000000.000000   

       price_per_unit    sale_amount__c   size_acres        size_sf  \
count      275.000000       1053.000000  1002.000000    1169.000000   
mean   3631771.927309  197379479.648299    57.384248  235702.285252   
std   11089755.850282  778643943.192584  1298.421962  259047.324484   
min        127.030000          0.000000    -1.000000       0.000000   
25%               nan               nan          nan            nan   
50%               nan               nan          nan            nan   
75%               nan               nan          nan            nan   
max   79583333

In [40]:
properties.isnull().sum()

id_props                     0
building_status           8217
building_tax_expenses     8743
id_deals                     0
city                         0
class                     6948
county                       0
sale_status                193
portfolio                    0
market                       1
num_buildings             7550
num_floors                4672
num_parking_spaces       10468
occupancy_rate            4052
price_per_sq_ft           6437
price_per_unit           12621
property_type_1            311
region__c                    5
sale_amount__c            6440
sale_date__c              6093
size_acres                4354
size_sf                   1568
size_units               10059
year_built                1858
dtype: int64

In [41]:
#RIET ANALYSIS
#87 unique accounts baiught 1178 properties  1110 unique 
#DO ACCOUNT WISE ANALYSIS
#DO PROPERTY ANALYSIS
#grouped_by = account_riet.groupby('id_accs')
#grouped_by.get_group('0012A000023XmOxQAK')


#accounts_per_property.describe() -> GIves Better Snapshot of data


    

account_riet['class'].value_counts()
#Class A    387
#Class B    275
#Class C     26
account_riet['city'].value_counts()
#City-0936    71
#City-0488    43   ->Need cleaning
#City-0090    30
account_riet['county'].value_counts()
#County-0284    90
#County-0172    62 ->need cleaning
#County-0160    48
account_riet['sale_status'].value_counts()
#N    1135  >bool conversion
#Y      43
account_riet['portfolio'].value_counts()
#False    633
#True     545
account_riet['market'].value_counts()
#Market-0059    120
#Market-0106    106
#Market-0047     89
account_riet['building_status'].fillna("NAN").value_counts()
#Existing              686
#NAN                   485
#Demolished              3
#Proposed                2
#Under Construction      2

account_riet['property_type_1'].fillna("NAN").value_counts()

# Retail                     411
# Office                     350
# Multi-Housing              203
# Industrial                 124
# Medical Office Building     37
# Land                        16
# Hotel-Lodging               12
# Self-Storage                 9
# Biotech                      5
# Seniors Housing              4
# Other                        4
# NAN                          3

account_riet['region__c'].fillna("NAN").value_counts()
# Southwest       310
# Southeast       254
# West            189
# Midwest         159
# Mid-Atlantic    144
# Northeast       122


Southwest       310
Southeast       254
West            189
Midwest         159
Mid-Atlantic    144
Northeast       122
Name: region__c, dtype: int64

In [42]:
account_indi = accounts_opportunities_properties[accounts_opportunities_properties['investor_type']==0]
account_indi.describe()

/home/akash_teehnoge/anaconda3/lib/python3.5/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


active_deals  activity_count  investor_type  num_deals_as_client  \
count   4769.000000     4769.000000    4769.000000          4769.000000   
mean      44.648773    20681.404068       0.000000            39.685888   
std       60.149364    24768.139247       0.000000            58.810435   
min        1.000000        0.000000       0.000000             1.000000   
25%       12.000000     5038.000000       0.000000             9.000000   
50%       25.000000    13334.000000       0.000000            20.000000   
75%       52.000000    29759.000000       0.000000            46.000000   
max      342.000000   193100.000000       0.000000           333.000000   

       num_deals_as_investor  number_of_properties  number_of_related_deals  \
count            4769.000000           4769.000000              4769.000000   
mean                4.962885             37.053051                68.608094   
std                 4.915401             53.073785               100.704266   
min                 0.000000              0.000000                 1.000000   
25%                 2.000000              8.000000                19.000000   
50%                 3.000000             18.000000                38.000000   
75%                 7.000000             43.000000                75.000000   
max                33.000000            442.000000               584.000000   

       number_of_related_properties  number_of_won_deals_as_client  \
count                   4769.000000                    4769.000000   
mean                      43.665129                      36.597819   
std                       56.870357                      57.908457   
min                        1.000000                       1.000000   
25%                       11.000000                       8.000000   
50%                       24.000000                      18.000000   
75%                       54.000000                      37.000000   
max                      449.000000                     330.000000   

       best_initial_bid  debt_yield  fiscalquarter  fiscalyear  num_buildings  \
count       4528.000000 4040.000000    4769.000000 4769.000000    3408.000000   
mean     4715598.289755    1.169770       2.430279 2011.669323       8.449531   
std     27001152.997187    3.351590       1.112304    3.851303      15.812830   
min            0.000000    0.000000       1.000000 2002.000000       0.000000   
25%                 nan         nan       1.000000 2009.000000            nan   
50%                 nan         nan       2.000000 2013.000000            nan   
75%                 nan         nan       3.000000 2015.000000            nan   
max    529400000.000000   17.300000       4.000000 2016.000000     375.000000   

       num_floors  num_parking_spaces  occupancy_rate   price_per_sq_ft  \
count 4010.000000         2458.000000     4139.000000       3638.000000   
mean     4.568080          400.406835       88.479101     369391.013263   
std      6.435898          378.375088       19.547806   18461109.417572   
min      0.000000            1.000000        0.000000          2.890000   
25%           nan                 nan             nan               nan   
50%           nan                 nan             nan               nan   
75%           nan                 nan             nan               nan   
max     60.000000         4802.000000      100.000000 1110000000.000000   

         price_per_unit    sale_amount__c     size_acres        size_sf  \
count       2022.000000       3717.000000    4102.000000    4589.000000   
mean     8143266.481869  116600592.374636     605.458402  205809.654519   
std    144591867.301364  437101134.671671   19592.974789  208776.078457   
min           49.230000          0.000000       0.000000       0.000000   
25%                 nan               nan            nan            nan   
50%                 nan               nan            nan            nan   
75%                 nan               nan            nan 

In [43]:
account_indi['property_type_1'].fillna("NAN").value_counts()
# Multi-Housing              1870
# Office                     1167
# Retail                      749
# Industrial                  465
# Hotel-Lodging               240
# Self-Storage                 65
# Seniors Housing              62
# Medical Office Building      45
# Land                         45
# Other                        31
# NAN                          24
# Biotech                       6




Multi-Housing              1870
Office                     1167
Retail                      749
Industrial                  465
Hotel-Lodging               240
Self-Storage                 65
Seniors Housing              62
Land                         45
Medical Office Building      45
Other                        31
NAN                          24
Biotech                       6
Name: property_type_1, dtype: int64